### PR TITLE
Updated delete member UI to add toggle to cancel subscriptions

### PR DESCRIPTION
**Pull request by allouis**
no-issue

This adds a new adapter for the member model so we can (optionally) add a `cancel` query parameter to the DELETE URL for members

Also updates the delete member modal to include a conditionally (based on active subscriptions) displayed toggle to enable/disable the `cancel` query parameter.

### CODE REVIEW

File: app/components/modal-delete-member.js [hunk 1]
```diff
@@ -1,25 +1,46 @@
 import ModalComponent from 'ghost-admin/components/modal-base';
 import {alias} from '@ember/object/computed';
+import {computed} from '@ember/object';
 import {inject as service} from '@ember/service';
 import {task} from 'ember-concurrency';
 
 export default ModalComponent.extend({
     membersStats: service(),
 
+    cancelSubscriptions: false,
```
Comment (kevinansfield) [inline app/components/modal-delete-member.js:10]: Can we change this to `shouldCancelSubscriptions`? I was tripped up thinking we were passing a function/action around when reading later code 😄

File: app/components/modal-delete-member.hbs [hunk 2]
```diff
@@ -7,6 +7,21 @@
...
+
+    {{#if this.hasActiveStripeSubscriptions}}
+        <div class="flex justify-between">
+            <div>
+                <h4>Cancel Stripe subscription</h4>
+                <p class="pa0 ma0">If disabled, member's Stripe subscription will continue</p>
+            </div>
+            <div class="for-switch">
+                <label class="switch" >
+                    <input type="checkbox" checked={{this.cancelSubscriptions}} class="gh-input" onclick={{action "toggleCancelSubscriptions"}}>
```
Comment (kevinansfield) [inline app/components/modal-delete-member.hbs:19]: `onclick=` is kinda deprecated now. We should use element modifiers wherever possible, in this case the [`{{on}}` modifier](https://guides.emberjs.com/release/components/template-lifecycle-dom-and-modifiers/#toc_event-handlers)

File: app/components/modal-delete-member.js (hunk 1 above)
Comment (allouis) [reply app/components/modal-delete-member.js:10]: Yes! Good point :relaxed:

File: app/components/modal-delete-member.hbs (hunk 2 above)
Comment (allouis) [reply app/components/modal-delete-member.hbs:19]: :+1: Thanks

**Pull request merged**